gcp-client/src/main/java/com/google/graphite/platforms/plugin/client/ComputeClient.java

```diff
@@ -0,0 +1,541 @@
+   * @throws IOException If an error occured in snapshot creation.
+   * @throws InterruptedException If snapshot creation is interrupted.
+   */
+  public void createSnapshotForDisk(String projectId, String zone, String diskId)
+      throws IOException, InterruptedException {
+    Snapshot snapshot = new Snapshot();
+    snapshot.setName(diskId);
+
+    Operation op = compute.disks().createSnapshot(projectId, zone, diskId, snapshot).execute();
+    // poll for result
+    waitForOperationCompletion(
+        projectId, op.getName(), op.getZone(), SNAPSHOT_TIMEOUT_MILLISECONDS);
+  }
+
+  /**
+   * Deletes persistent disk snapshot. Does not block.
+   *
+   * @param projectId Google cloud project id.
+   * @param snapshotName Name of the snapshot to be deleted.
+   * @throws IOException If an error occurred in deleting the snapshot.
+   */
+  public void deleteSnapshot(String projectId, String snapshotName) throws IOException {
+    compute.snapshots().delete(projectId, snapshotName).execute();
+  }
+
+  /**
+   * Returns snapshot with name snapshotName
+   *
+   * @param projectId Google cloud project id.
+   * @param snapshotName Name of the snapshot to get.
+   * @return Snapshot object with given snapshotName. Null if not found.
+   * @throws IOException If an error occurred in retrieving the snapshot.
+   */
+  public Snapshot getSnapshot(String projectId, String snapshotName) throws IOException {
+    SnapshotList response;
+    Compute.Snapshots.List request = compute.snapshots().list(projectId);
+
+    do {
+      response = request.execute();
+      if (response.getItems() == null) {
+        continue;
+      }
+      for (Snapshot snapshot : response.getItems()) {
+        if (snapshotName.equals(snapshot.getName())) return snapshot;
+      }
+      request.setPageToken(response.getNextPageToken());
+    } while (response.getNextPageToken() != null);
+
+    return null;
+  }
+
+  /**
+   * Appends metadata to an instance. Any metadata items with existing keys will be overwritten.
+   * Otherwise, metadata is preserved. This method blocks until the operation completes.
+   *
+   * @param projectId
+   * @param zone
+   * @param instanceId
+   * @param items
+   * @throws IOException
+   * @throws InterruptedException
+   */
+  public Operation.Error appendInstanceMetadata(
+      String projectId, String zone, String instanceId, List<Metadata.Items> items)
+      throws IOException, InterruptedException {
+    zone = nameFromSelfLink(zone);
+    Instance instance = getInstance(projectId, zone, instanceId);
+    Metadata existingMetadata = instance.getMetadata();
+
+    List<Metadata.Items> newMetadataItems = mergeMetadataItems(items, existingMetadata.getItems());
+    existingMetadata.setItems(newMetadataItems);
+
+    Operation op =
+        compute.instances().setMetadata(projectId, zone, instanceId, existingMetadata).execute();
+    return waitForOperationCompletion(projectId, op.getName(), op.getZone(), 60 * 1000);
```

Comment:
consider creating an override for this call that takes a vanilla operation rather than requiring each caller to break out .getName() and getZone()

Reply:
Done